include/git2/status.h

```diff
@@ -173,12 +173,17 @@ typedef enum {
  * The `pathspec` is an array of path patterns to match (using
  * fnmatch-style matching), or just an array of paths to match exactly if
  * `GIT_STATUS_OPT_DISABLE_PATHSPEC_MATCH` is specified in the flags.
+ *
+ * The `tree` is a tree to be used instead of the head commit's tree for
+ * comparing with the index. For example, this can be useful when preparing
+ * to do an amend commit.
  */
 typedef struct {
 	unsigned int      version;
 	git_status_show_t show;
 	unsigned int      flags;
 	git_strarray      pathspec;
+	git_tree          *tree;
```

Comment:
We use the term `baseline` for the overrideable `HEAD` in `git_checkout`.  I wonder if `baseline` would be more consistent here, too?